bokehjs/src/lib/core/view.ts

```diff
@@ -129,3 +144,58 @@ export class View implements ISignalable {
     }
   }
 }
+
+export class ViewManager {
+  readonly roots: Set<View>
+
+  constructor(roots: View[] = []) {
+    this.roots = new Set(roots)
+  }
+
+  get<T extends HasProps>(model: T): ViewOf<T> | null {
+    for (const view of this.roots) {
+      if (view.model == model)
+        return view
+    }
+    return null
+  }
+
+  add(view: View): void {
+    this.roots.add(view)
+  }
+
+  delete(view: View): void {
+    this.roots.delete(view)
+  }
+
+  *[Symbol.iterator](): IterViews {
+    yield* this.roots
+  }
+
+  *find<T extends HasProps>(model: T): IterViews<ViewOf<T>> {
+    function *descend(view: View): IterViews {
+      if (view.model == model) {
+        yield view
+      } else {
+        for (const child of view.children()) {
+          yield* descend(child)
+        }
+      }
+    }
+
+    for (const root of this.roots) {
+      yield* descend(root)
+    }
+  }
+
+  find_one<T extends HasProps>(model: T): ViewOf<T> | null {
+    for (const view of this.find(model)) {
+      return view
+    }
+    return null
+  }
+
+  find_all<T extends HasProps>(model: T): ViewOf<T>[] {
+    return [...this.find(model)]
+  }
+}
```

Comment:
I really like this, it's long overdue. I do think this class is amenable to basic unit testing